### PR TITLE
Adds emagging to mechs

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Allows mechs to be emagged.
Emagging will disable a mechs equipment for 15 seconds and prevent it from being DNA locked (also unlocking DNA locked mechs).

## Why It's Good For The Game

Adds emaggability to something that should really have it. Allows traitors to counter validhunting mechs by disabling them for 15 seconds and destroying it, or allows them to hijack unpiloted, unprotected mechs.

## Changelog
:cl:
add: Allows mechs to be emagged.
/:cl:

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -466,6 +466,19 @@
 				speech_bubble_recipients.Add(M.client)
 		INVOKE_ASYNC(GLOBAL_PROC, /proc/flick_overlay, image('icons/mob/talk.dmi', src, "machine[say_test(raw_message)]",MOB_LAYER+1), speech_bubble_recipients, 30)
 
+/obj/mecha/emag_act()
```
Comment (PowerfulBacon) [inline code/game/mecha/mecha.dm]:
```suggestion
/obj/mecha/emag_act(mob/user)
```

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -466,6 +466,19 @@
...
+	if(obj_flags & EMAGGED)
+		to_chat(user, "<span class='warning'>The mech suit's internal controls are damaged beyond repair!</span>")
+		return
+	obj_flags |= EMAGGED
+	playsound(src, "sparks", 100, 1)
+	to_chat(user, "<span class='warning'>You short out the mech suit's internal controls.</span>")
+	dna_lock = null
+	equipment_disabled = TRUE
+	log_message("System emagged detected", LOG_MECHA, color="red")
+	addtimer(CALLBACK(src, /obj/mecha/proc/restore_equipment), 15 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)
```
Comment (qwertyquerty) [inline code/game/mecha/mecha.dm:480]: If I were you I wouldn't make the timer override or they could just permastun by continually clicking and the mech would never have a chance of freedom

Comment (PowerfulBacon) [reply code/game/mecha/mecha.dm:480]: but you can't emag it twice

**Review by qwertyquerty — APPROVED** [review]

**Comment by githubuser4141** [comment]
I'm not a coder, but does this work for a ninja's katana..? If so then maybe it should have a cooldown or something, otherwise a mech will basically have no chance against a ninja who can completely incapacitate a mech with one stab.

**Pull request merged**